honeybee_vtk/file_writers.py

```diff
@@ -0,0 +1,229 @@
+    """
+
+    # Create a new folder at target folder to store all the files
+    folder_name = 'hbjson-html'
+    target_folder = os.path.join(target_folder, folder_name)
+    if os.path.exists(target_folder):
+        shutil.rmtree(target_folder)
+
+    # Setting DataSet Writer and file extension
+    vtk_writer = vtk.vtkJSONDataSetWriter()
+    vtk_extension = ""
+    
+    # Get points and face_types from HBJSON
+    points, hb_types = read_hbjson(hbjson)
+
+    # Get points grouped for each Honeybee face_type
+    grouped_points = group_by_face_type(points, hb_types)
+
+    # Write VTK files based on Honeybee face_type and Honeybee object type
+    for hb_type in grouped_points:
+        write_polydata(grouped_points[hb_type], hb_type, vtk_writer, vtk_extension,
+                       target_folder)
+
+    # Names of VTK files to be written
+    file_names = list(grouped_points.keys())
+
+    # write grid points to a vtk file if grids are found in HBJSON
+    grids = check_grid(hbjson)
+
+    # If grids are there in HBJSON and they are requested
+    if grids and include_grids:
+        grid_file_names = _write_grids(grids, vtk_writer, vtk_extension, target_folder)
+        file_names.extend(grid_file_names)
+
+    # Write vectors if they are requested
+    if include_vectors:
+        to_write_vectors = [hb_types, grouped_points, grids, include_grids, vtk_writer,
+                            vtk_extension, target_folder]
+        vector_file_names = _write_vectors(*to_write_vectors)
+        file_names.extend(vector_file_names)
+
+    # Write index.json
+    write_index_json(target_folder, file_names)
+
+    # Collect everything in a .zip
+    # Use the name of HBJSON if file name is not provided
+    if not file_name:
+        name = '.'.join(os.path.basename(file_path).split('.')[:-1])
+        file_name = clean_string(name)
+
+    # remove extension if provided by user
+    file_name = file_name if not file_name.lower().endswith('.zip') else file_name[:-4]
+
+    # Create a .zip file to capture all the generated .vtk files
+    zip_name = file_name
+    file_path = os.path.join(target_folder, file_name)
+    shutil.make_archive(file_path, 'zip', target_folder)
+ 
+    # Delete the index.json if there is permission
+    try:
+        file_name = os.path.join(target_folder, 'index.json')
+        os.remove(file_name)
+    except OSError:
+        warnings.warn(
+            'Honeybee does not have permission to delete index.json.'
+            ' You may please delete the .vtk files manually.'
+        )
+    # Delete the folders containing json datasets if there is permission
+    for file_name in file_names:
+        try:
+            file_name = os.path.join(target_folder, file_name)
+            shutil.rmtree(file_name)
+        except OSError:
+            warnings.warn(
+                f'Honeybee does not have permission to delete {file_names}.'
+                ' You may please delete the .vtk files manually.'
+            )
```

Comment:
This part of the code is shared between the two function. I would make it a separate `_writer` function and re-use for both functions. Right now it looks fine but over time it will be a pain to maintain and refactor.

Reply:
Yes. There is some duplication and an opportunity to capture in a function. I will do this later.